include/sys/bus.h

```diff
@@ -184,37 +180,18 @@ static inline resource_t *bus_alloc_resource(device_t *dev, res_type_t type,
                                              int rid, rman_addr_t start,
                                              rman_addr_t end, size_t size,
                                              res_flags_t flags) {
-  return BUS_DRIVER(dev)->bus.alloc_resource(dev, type, rid, start, end, size,
-                                             flags);
+  return BUS_METHOD(dev->parent)
+    .alloc_resource(dev, type, rid, start, end, size, flags);
 }
 
 #define bus_alloc_irq(dev, rid, irq, flags)                                    \
   bus_alloc_resource((dev), RT_IRQ, (rid), (irq), (irq), 1, (flags))
```

Comment:
Should we remove this define? It is not used anywhere.

Reply:
Done.